ARGO/src/EnemyManager.cpp

```diff
@@ -0,0 +1,150 @@
+#include "stdafx.h"
+#include "EnemyManager.h"
+
+EnemyManager::EnemyManager(SDL_Renderer* t_renderer, float t_initialSpawnDelay, EventManager& t_eventManager, PhysicsSystem& t_physicsSystem, CollisionSystem& t_collisionSystem, HealthSystem& t_healthSystem, AiSystem& t_aiSystem, RenderSystem& t_renderSystem, LevelManager& t_levelManager) :
+	m_nextEnemy(0),
+	m_renderer(t_renderer),
+	m_spawnEnemyTimer(t_initialSpawnDelay),
+	m_physicsSystem(t_physicsSystem),
+	m_collisionSystem(t_collisionSystem),
+	m_healthSystem(t_healthSystem),
+	m_aiSytem(t_aiSystem),
+	m_renderSystem(t_renderSystem),
+	m_levelManager(t_levelManager)
+{
+}
+
+void EnemyManager::init()
+{
+	m_audioMgr = AudioManager::Instance();
+
+	for (auto& enemy : m_enemies)
+	{
+		enemy.addComponent(new TransformComponent());
+		enemy.addComponent(new AiComponent(AITypes::eMelee, AIStates::eWander, 15.0f, 1.0f));
+		enemy.addComponent(new ColliderCircleComponent(Utilities::ENEMY_RADIUS));
+		enemy.addComponent(new TagComponent(Tag::Enemy));
+		enemy.addComponent(new ForceComponent());
+		enemy.addComponent(new HealthComponent(Utilities::ENEMY_HP, 0));
+		enemy.addComponent(new VisualComponent("EnemyFast.png", m_renderer));
+	}
+}
+
+void EnemyManager::update(float t_dt)
+{
+	spawnGroup(t_dt);
+	for (auto& enemy : m_enemies)
+	{
+		HealthComponent* hpComp = static_cast<HealthComponent*>(enemy.getComponent(ComponentType::Health));
+		if (hpComp && hpComp->isAlive())
+		{
+			m_healthSystem.update(enemy, t_dt);
+			m_physicsSystem.update(enemy, t_dt);
+			m_collisionSystem.update(enemy);
+			m_aiSytem.update(enemy);
+		}
+	}
+}
+
+void EnemyManager::spawnGroup(float t_dt)
+{
+	m_spawnEnemyTimer -= t_dt;
+	if (m_spawnEnemyTimer <= 0)
+	{
+		m_spawnEnemyTimer = SPAWN_ENEMY_RATE;
+		Entity* spawnTile = nullptr;
+		int attempts = 0;
```

Comment:
what does attempts do

Reply:
Okay you can very clearing tell it's counting the amount of attempts for the while loop.